src/wazuh_modules/wm_exec.c

```diff
@@ -298,54 +302,49 @@ int wm_exec(char *command, char **output, int *exitcode, int secs)
         // Parent
 
         wm_append_sid(pid);
+        close(pipe_fd[1]);
+        tinfo.pipe = pipe_fd[0];
 
-        if (output) {
-            close(pipe_fd[1]);
-            tinfo.pipe = pipe_fd[0];
-
-            // Launch thread
+        // Launch thread
 
-            pthread_mutex_lock(&tinfo.mutex);
+        pthread_mutex_lock(&tinfo.mutex);
 
-            if (pthread_create(&thread, NULL, reader, &tinfo)) {
-                merror("Couldn't create reading thread.");
-                pthread_mutex_unlock(&tinfo.mutex);
-                return -1;
-            }
+        if (pthread_create(&thread, NULL, reader, &tinfo)) {
+            merror("Couldn't create reading thread.");
+            pthread_mutex_unlock(&tinfo.mutex);
+            return -1;
+        }
 
-            gettime(&timeout);
-            timeout.tv_sec += secs;
+        gettime(&timeout);
+        timeout.tv_sec += secs;
 
-            // Wait for reading termination
+        // Wait for reading termination
 
-            switch (secs ? pthread_cond_timedwait(&tinfo.finished, &tinfo.mutex, &timeout) : pthread_cond_wait(&tinfo.finished, &tinfo.mutex)) {
-            case 0:
-                retval = 0;
-                break;
+       switch (secs ? pthread_cond_timedwait(&tinfo.finished, &tinfo.mutex, &timeout) : pthread_cond_wait(&tinfo.finished, &tinfo.mutex)) {
```

Comment:
This part has no sense since we have not created a pipe. This thread may fail and return immediately, so the timeout could not be computed.